src/main/flight/mixer.c

```diff
@@ -726,7 +726,11 @@ static void applyMixToMotors(float motorMix[MAX_SUPPORTED_MOTORS], motorMixer_t
     // Now add in the desired throttle, but keep in a range that doesn't clip adjusted
     // roll/pitch/yaw. This could move throttle down, but also up for those low throttle flips.
     for (int i = 0; i < motorCount; i++) {
-        float motorOutput = motorOutputMin + (motorOutputRange * (motorOutputMixSign * motorMix[i] + throttle * activeMixer[i].throttle));
+        float motorOutput = pidApplyThrustLinearization(
```

Comment:
This looks a bit confusing if `USE_THRUST_LINEARIZATION` is not defined - maybe wrap `pidApplyThrustLinearization()` and `pidCompensateThrustLinearization()` in conditionals.
￼

Reply:
My view was that it's better to limit the amount of conditionals throughout the code. It seems like we're at a point where almost any code change forces us to be selective wrt board support. But do we really want every line of source be wrapped in conditionals if it is avoidable?

Reply:
Not sure of this. Yes, making every feature optional at compile time adds a lot of complexity. But in my opinion, hiding the complexity (and forcing the reader of the code to search for it, or overlook it) does not do anything to improve this situation.
In the case at hand, a reader of `mixer.c` file will not know per se that `pidApplyThrustLinearization()` yields identity (or doesn't even exist if we make these conditional) in the case that `USE_THRUST_LINEARIZATION` isn't defined, so adding an explicit conditional actually increases the local readability by making this clear.

Reply:
I think the right question to ask is what we'd like the source to look if we didn't have memory constraints. Would we still like to sprinkle conditionals throughout the code to make the casual reader of the source aware that a line belongs to a certain feature or not?

If the answer to this question is no then I don't see how adding a conditional can have any value when the casual reader of mixer.c will have no knowledge of the memory constraints the code in pid.c faced.

One might also consider that most of the features that get called using the applyXXX paradigm can get turned off in the config. If one considers conditionals essential for clear readability of compile time enabled features then surely the config should get queried before calling these functions? I think not.

Reply:
> I think the right question to ask is what we'd like the source to look if we didn't have memory constraints.

I don't see any point in asking purely hypothetical questions, and then distorting the reality to match the purely hypothetical answers.

> Would we still like to sprinkle conditionals throughout the code to make the casual reader of the source aware that a line belongs to a certain feature or not?

Yes. Forcing the casual reader to dig through multiple files only to find out that this code hardcodes to yielding identity with the compile time settings he uses is not making the source more readable.

Reply:
I personally find the already existing conditionals in the source to be highly annoying and somewhat impeding my ability to quickly read the source. And that impediment I consider more severe than having to check whether a feature is conditional at the actual place of implementation if it's not working.

But your call. I'll add them back in.

Reply:
You know what, I really do think it's wrong to sprinkle the conditionals. That will pollute the source code to no end going forward. How about pidCompensateThrustLinearizationIfSupported()?

Reply:
But yeah - I see that you've done it the way you suggested also in most other cases. So maybe the ifSupported pattern would scale? For me the more code I can take in at a glance the quicker I can understand it. Anything that's just noise when normally looking at the code thus has a "maintenance cost".

Reply:
@joelucid: If you don't want this merged, just close it. No point dicking around and wasting the time of people caring to review your code.

Reply:
@mikeller how about keeping the tone professional?

Reply:
@joelucid: What about not wasting other people's time by adding commits for ridiculous proposals?

If we were to change every one of the 100s of functions that is currently using the 'if a function is conditional, then calls to this function should be conditional as well' into `<function name>IfSupported()` (or rather `<function name>If<conditional>IsDefined()` to be correct and provide the same amount of information as the way the code currently is) then the code would become a lot more unreadable in my opinion.

Reply:
I'll revert to the previous commit. But I think you managed to express pretty much exactly how I feel about the use of adding two lines of code to a call of any function that might not be supported on some boards.

I'm sorry you feel I'm wasting your time. If that's a concern shared by more people here I'll gladly take the features currently under review/development and make them available in more welcoming environments.

Reply:
Thanks. As I stated before, I like the innovativeness of the features that you contribute, and I'd hate to see you take it somewhere else. I don't know why our discussions are turning sour so often, but my apologies for being party in that.

Can I also get you to rebase this and squash into one commit please?